File: ruby/pangram/pangram.rb
class Pangram
VERSION = 1

  def self.is_pangram?(sentance)
    sentance = sentance.gsub(/\W+/, '').downcase.chars
    alphabet(sentance).all?
  end

  def self.alphabet(sentance)
    alphabet = ("a".."z").to_a
    alphabet.map do |x|
      sentance.include?(x)
    end.uniq
  end
end

File: javascript/pangram/pangram.js
function Pangram(sentance) {
  this.sentance = sentance.toLowerCase();
}

Pangram.prototype.isPangram = function(){
  var alphabet = 'abcdefghijklmnopqrstuvwxyz'.split('');

  for (var i = 0; i < alphabet.length; i++) {
    if(this.sentance.indexOf(alphabet[i]) === -1){
      return false
    }
  }
  return true

}

module.exports = Pangram;
